app/main.py

```diff
@@ -1,12 +1,26 @@
+import hashlib
+
 from flask import abort, Flask, jsonify, request
 
-from app.conf import DEBUG
-from app.memorize import CacheMissError
+from app import conf
+from app.util import make_queue
+from app.memcached import memcached
 from app.search.recommendation import SearchRecommendation
 
 
 application = Flask(__name__)
-application.config['DEBUG'] = DEBUG
+application.config.update(
+    CELERY_BROKER_URL=conf.CELERY_BROKER_URL,
+    DEBUG=conf.DEBUG
+)
+queue = make_queue(application)
+
+
+@queue.task(name='main.recommend')
+def recommend(q, key):
+    recommendation = SearchRecommendation(q).do_search(q)
```

Comment:
Could you explain in a bit more detail how cache invalidation is managed in this code? Maybe in the code docs?

There will be a period of time between invalidating a cached result and a fresh response being inserted into the cache. Do we do anything to avoid creating redundant jobs during that period of time? If not, the number of redundant API requests will grow with the number of users, so I guess we can ignore it for now, but might be best to be explicit about that, and file a followup bug. (More likely, there's a caching layer someplace that I'm missing :-)